### PR TITLE
Xenomorphs can play basketball.

### PR DESCRIPTION
**Pull request by Spyroshark**
:cl: Spyroshark and Drawsstuff
add: Xenomorphs can now play basketball.
/:cl:

xenobasketball

### CODE REVIEW

File: code/game/machinery/computer/HolodeckControl.dm [hunk 1]
```diff
@@ -399,6 +399,18 @@ var/global/list/holodeck_programs = list(
...
+	//Can be picked up by aliens
+/obj/item/toy/beach_ball/holoball/attack_paw(user as mob)
+	if(isxeno(user))
+		attack_alien(user)
+
+/obj/item/toy/beach_ball/holoball/attack_alien(mob/living/carbon/Xenomorph/user)
+		attack_hand(user)
+
+/obj/item/toy/beach_ball/holoball/attack_hand(mob/user)
+	return ..()
```
Comment (DominikPanic) [inline code/game/machinery/computer/HolodeckControl.dm]: This shouldn't be necessary as in this state it does nothing.

File: code/game/machinery/computer/HolodeckControl.dm [hunk 2]
```diff
@@ -399,6 +399,14 @@ var/global/list/holodeck_programs = list(
 	desc = "Here's your chance, do your dance at the Space Jam."
 	w_class = 4 //Stops people from hiding it in their bags/pockets
 
+	//Can be picked up by aliens
+/obj/item/toy/beach_ball/holoball/attack_paw(user as mob)
+	if(isxeno(user))
+		attack_alien(user)
+
+/obj/item/toy/beach_ball/holoball/attack_alien(mob/living/carbon/Xenomorph/user)
+		attack_hand(user)
```
Comment (DominikPanic) [inline code/game/machinery/computer/HolodeckControl.dm]: Incorrect indentation.
```suggestion
	attack_hand(user)
```

File: code/game/machinery/computer/HolodeckControl.dm [hunk 3]
```diff
@@ -399,6 +399,14 @@ var/global/list/holodeck_programs = list(
 	desc = "Here's your chance, do your dance at the Space Jam."
 	w_class = 4 //Stops people from hiding it in their bags/pockets
 
+	//Can be picked up by aliens
+/obj/item/toy/beach_ball/holoball/attack_paw(user as mob)
+	if(isxeno(user))
```
Comment (DominikPanic) [inline code/game/machinery/computer/HolodeckControl.dm]: Prefer early returns instead of nested logic.

Comment (Rohesie) [reply code/game/machinery/computer/HolodeckControl.dm]: That means
```if(!isxeno(user))
	return FALSE
attack_alien(user)
```
The point to this is to diminish unnecessary indentation as procs grow complex.

**Comment by Spyroshark** [comment]
well what else would xenomorphs do when they aren't infecting people or fighting marines?

**Pull request merged**